deps(webrtc): don't include `webrtc` by default if `tokio` is enabled

altonen opened the pull request:
## Description

## Notes & open questions

Don't enable `webrtc` by default if `tokio` is enabled. Bug in cargo causes webrtc-related dependencies to be pulled even if `webrtc` is not enabled and causes issues downstream with incompatible dependencies.

<!--
Any notes, remarks or open questions you have to make about the PR which don't need to go into the final commit message.
-->

## Change checklist

<!-- Please add a Changelog entry in the appropriate crates and bump the crate versions if needed. See <https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

thomaseizinger commented on transports/webrtc/Cargo.toml:
You'll need to bump the version of this crate to `0.4.0-alpha.5`.

thomaseizinger reviewed (approved):
This looks good!

We can't really trust CI that much for backports because all sorts of stuff breaks with old Rust versions. Before we make the release, can you confirm that this PR works as intended with polkadot?

thomaseizinger reviewed (changes requested):
I am somewhat hesitant to make a breaking change (MSRV) in a patch release if the actual motivation for this is just working around a cargo bug :(

I'll withdraw my approval for now and I'd like to understand better, what is actually going on here.

altonen committed:
Don't include `webrtc` into the `tokio` feature

This works around a bug in cargo which unconditionally pulls the
dependencies even if the `webrtc` feature is not enabled and causes
issues in downstream with other conflicting dependencies.
